etc/config/asmx.properties

```diff
@@ -0,0 +1,4 @@
+compilers=AsmX&AsmX2
+
+group.AsmX2.baseName=AsmX
+compiler.AsmX2.exe=../../kernel.js
```

Comment:
can you ensure this file, and the others, end on a newline please? Thanks!

Reply:
I now realized that this file may need to be changed. I need to configure the compiler correctly, for example. It can Compile under ARM. (Perhaps in the future there will be compilation for other architectures) And the compiler should be called in a different way.
Example:
```
node kernel.js asmx-cli build arm InputFileName.asm Output FileName
```
I hope that it is possible to run Node JS.

Reply:
> I hope that it is possible to run Node JS.

_maybe_ -- what version of node does it need? what packages? each of our compilers is as separated as possible. that is, each compiler in `/opt/compiler-explorer/some-compiler-v1.2.3` contains pretty much everything needed to run it: all its dependencies etc.

We _do_ support running some interpreted languages like `python` which need a particular version installed, but it's quite a pain.

Essentially whatever we install needs to be runnable from a pretty bare Ubuntu 18.04 image (which is what our many compute nodes are running).

Reply:
> > I hope that it is possible to run Node JS.
> 
> _maybe_ -- what version of node does it need? what packages? each of our compilers is as separated as possible. that is, each compiler in `/opt/compiler-explorer/some-compiler-v1.2.3` contains pretty much everything needed to run it: all its dependencies etc.

v16.15.1, list packages in compiler file package.json.

Reply:
Ok , that's going to be a significant amount more work: we'lll need to support `npm install` packages in our infra, or something similar, as well as a way of running `node` itself.

Reply:
I imagine the simplest way to go here is to have the asmx stuff packaged into something like this https://github.com/vercel/pkg. This way we get a regular executable (but there is of course some overhead so you don't get the raw compiler speed (but does that really matter? we are running node anyways))